docs/DeveloperGuide.md

```diff
@@ -343,77 +343,208 @@ b. Require developers to download those libraries manually (this creates extra w
+`* * *` | user who have completed one or more events | view a reverse chronological log of all completed events | see which events I have marked as `done` in the past
+`* * *` | user who have dismissed a task | reverse chronological log of all dismissed tasks | see which tasks I have dismissed
+`* * *` | user who have completed or dismissed tasks or events | recycle a task or an event kept in the expired list | 
+`* * *` | user who have completed or dismissed tasks or events | clear all the past tasks and events from the expired list | eliminate no-longer-relevant tasks and events and keep the log clean
+`* * *` | user | see error message when I enter an invalid command | ammend the command appropriately
+`* * *` | user | undo the last command entered |
+`* * *` | user | switch between multiple storage files | have separate task managers of different aspects of my life
+`* * *` | user | specify the location of the storage file <br /><br /> **condition**:<br />If the file does not exist in the path, create a new file| place the file anywhere I find convenient, for example, in a shared folder
+`* * *` | advanced user | have raw access to the storage files | directly perform CRUD operation on the data all at once
+`* *` | user with recurring tasks or events| add a task or an event that are reccuring| avoid the trouble of manually adding repeatedly on a regular basis
+`* *` | user | identify free time slots in a day, week or month | can evaluate how busy I am and optimize the date for tasks and events to add
+`* *` | user | display daily, weekly, or monthly calendar views | visualize the deadlines of tasks and times of events for the day/week/month
+`* *` | user | view the history of all previously undertaken actions | prevent duplicate commands and correct entries that are erroneous in hindsight
+`* *` | user | choose between automatic save and manual save |
+`*` | user | have an auto-completion for commands that are predictable | avoid the trouble of typing in the full command
+`*` | user | enter the options for a command in any order | 
+
+## Appendix B : Sample Use Cases
+
+(For all use cases below, the **System** is the `TaskCrusher` and the **Actor** is the `user`, unless specified otherwise)
+
+
+#### Use case: Add an event
+
+**MSS**
+
+1. User enters the name of an event to be added, together with its start and end date/time and other options.
+2. TaskCrusher adds the event to the active list
+3. TaskCrusher displays back the details of the add action.
+
+Use case ends.
+
+**Extensions**
 
-{More to be added}
+2a. The command format is invalid
 
-## Appendix B : Use Cases
+> 2a1. TaskCrusher shows an error message <br>
+  Use case resumes at step 1
+
+2b. The entered time slot is already occupied by another event 
+> 2b1. TaskCrusher notifies the user that the time slot for the event has been already occupied by other event.
+  2b2. User decides whether or not to force the addition. 
```

Comment:
and asks for user confirmation (how else will the user know he can force add or not?)